Cubical/Foundations/GroupoidLaws.agda

```diff
@@ -0,0 +1,95 @@
+{-
+
+This file proves the higher groupoid structure of types
+
+-}
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Foundations.GroupoidLaws where
+
+open import Cubical.Core.Primitives
+open import Cubical.Core.Prelude
+
+private
+  variable
+    ℓ : Level
+    A : Set ℓ
+
+-- some useful notation (move to prelude?)
+
+_·_ :  {x y z : A} → (x ≡ y) → (y ≡ z) → (x ≡ z)
+x≡y · y≡z = compPath x≡y y≡z 
+
+infixr 30 _·_
+
+_⁻¹ : {x y : A} → (x ≡ y) → (y ≡ x)
+x≡y ⁻¹ = sym x≡y
+
+-- homogeneous groupoid laws
+
+symRefl : (x : A) →
+  Path (Path A x x) refl (refl ⁻¹)
+symRefl x i = refl 
+
+compPathRefl : (x : A) →
+  Path (Path A x x) refl (refl · refl)
+compPathRefl x j i =
+  hfill (λ j → λ { (i = i0) → x
+                  ; (i = i1) → x }) (inc x) j
+
+symInvo : (x y : A) (p : x ≡ y) →
+     Path (Path A x y) p ((p ⁻¹)⁻¹)
+symInvo x y p = refl
+
+rUnit : (x y : A) (p : x ≡ y) →
+  Path (Path A x y) p (p · refl)
+rUnit x y p j i =
+  hfill (λ j → λ { (i = i0) → x
+                 ; (i = i1) → y }) (inc (p i)) j
+
+lUnit : (x y : A) (p : x ≡ y) →
+  Path (Path A x y) p (refl · p)
+lUnit x y p k i =
+  hcomp (λ j → λ { (i = i0) → x
+                  ; (i = i1) → p (~ k ∨ j )
+                  ; (k = i0) → p i
+               -- ; (k = i1) → compPath-filler refl p j i
+                  }) (p (~ k ∧ i ))
+
+rCancel : (x y : A) (p : x ≡ y) →
+   Path (Path A x x) (p · (p ⁻¹)) refl
+rCancel x y p j i =
+  hcomp (λ k → λ { (i = i0) → x
+                  ; (i = i1) → p (~ k ∧ ~ j)
+               -- ; (j = i0) → hfill (λ w → λ { (i = i0) → x; (i = i1) → p (~ w) }) (inc (p i)) k
+                  ; (j = i1) → x }) (p (i ∧ ~ j))
+
+lCancel : (x y : A) (p : x ≡ y) →
+   Path (Path A y y) refl ((p ⁻¹) · p)
+lCancel x y p k i =
```

Comment:
I suggest making `(x y : A)` implicit. Also you can do

```
lCancel x y p i = rCancel y x (p ⁻¹) (~ i)
```

instead of a complicated `hcomp`.

Reply:
That's really nice!